IPython/core/magics/osm.py

```diff
@@ -481,6 +520,7 @@ def dhist(self, parameter_s=''):
 
         dh = self.shell.user_ns['_dh']
         if parameter_s:
+            args = []
```

Comment:
This empty `args` is never used

Reply:
Yes, it's overwritten on line [525](https://github.com/ipython/ipython/pull/11365/files#diff-e72931d7a47ab4a04ef88193e513a39fR525) and the `return` on line [528](https://github.com/ipython/ipython/pull/11365/files#diff-e72931d7a47ab4a04ef88193e513a39fR528) prevents a case where `args` is not in scope. 

However, the linter wasn't fully aware of this and got worried that `args` may not be implemented at the time of use. Looking forward, if the except is changed to handle a specific exception, rather than a generic exception, then this will prevent `args` from being out of scope.

This may be over protective, so I can still remove the line in question if preferred.